test/e2e/Core_selection.spec.js

```diff
@@ -1729,6 +1729,213 @@ describe('Core_selection', () => {
+        |   ║   :   :   :   :   |
+        `).toBeMatchToSelectionPattern();
+
+        // Area selection (headers are selected).
+        $(getCell(1, 1)).simulate('mousedown');
+        $(getCell(4, 4)).simulate('mouseover');
+        $(getCell(4, 4)).simulate('mouseup');
+
+        expect(getSelected()).toEqual([[1, 1, 4, 4]]);
+        expect(getSelectedRangeLast().highlight.row).toBe(1);
+        expect(getSelectedRangeLast().highlight.col).toBe(1);
+        expect(getSelectedRangeLast().from.row).toBe(1);
+        expect(getSelectedRangeLast().from.col).toBe(1);
+        expect(getSelectedRangeLast().to.row).toBe(4);
+        expect(getSelectedRangeLast().to.col).toBe(4);
+        expect(`
+        |   ║   :   :   :   :   |
+        |===:===:===:===:===:===|
+        |   ║   :   :   :   :   |
+        |   ║   :   :   :   :   |
+        |   ║   :   :   :   :   |
+        |   ║   :   :   :   :   |
+        |   ║   :   :   :   :   |
+        `).toBeMatchToSelectionPattern();
+
+        // Column header selection.
+        simulateClick($(getCell(-1, 1)), 'LMB');
+
+        expect(getSelected()).toEqual([[-1, 1, 4, 1]]);
+        expect(getSelectedRangeLast().highlight.row).toBe(0);
+        expect(getSelectedRangeLast().highlight.col).toBe(1);
+        expect(getSelectedRangeLast().from.row).toBe(-1);
+        expect(getSelectedRangeLast().from.col).toBe(1);
+        expect(getSelectedRangeLast().to.row).toBe(4);
+        expect(getSelectedRangeLast().to.col).toBe(1);
+        expect(`
+        |   ║   : * :   :   :   |
+        |===:===:===:===:===:===|
+        |   ║   :   :   :   :   |
+        |   ║   :   :   :   :   |
+        |   ║   :   :   :   :   |
+        |   ║   :   :   :   :   |
+        |   ║   :   :   :   :   |
+        `).toBeMatchToSelectionPattern();
+
+        // Row header selection.
+        simulateClick($(getCell(1, -1)), 'LMB');
+
+        expect(getSelected()).toEqual([[1, -1, 1, 4]]);
+        expect(getSelectedRangeLast().highlight.row).toBe(1);
+        expect(getSelectedRangeLast().highlight.col).toBe(0);
+        expect(getSelectedRangeLast().from.row).toBe(1);
+        expect(getSelectedRangeLast().from.col).toBe(-1);
+        expect(getSelectedRangeLast().to.row).toBe(1);
+        expect(getSelectedRangeLast().to.col).toBe(4);
+        expect(`
+        |   ║   :   :   :   :   |
+        |===:===:===:===:===:===|
+        |   ║   :   :   :   :   |
+        | * ║   :   :   :   :   |
+        |   ║   :   :   :   :   |
+        |   ║   :   :   :   :   |
+        |   ║   :   :   :   :   |
+        `).toBeMatchToSelectionPattern();
+      });
+    });
+
+    describe('is set for single cell/column', () => {
+      it('to value `true`', () => {
+        handsontable({
+          rowHeaders: true,
+          colHeaders: true,
+          cells(row) {
+            const cellProperties = {};
+
+            if (row === 1) {
+              cellProperties.disableVisualSelection = true;
+            }
```

Comment:
Giant nitpick as this is just a test (and do we even do style comments in this repo?), maybe use this initialization method? (changing the shape of the object might be less performant? https://blog.bitsrc.io/the-jit-in-javascript-just-in-time-compiler-798b66e44143)

```suggestion
            const cellProperties = {
              disableVisualSelection: row === 1 ? true : undefined
            };
```
```suggestion
            const cellProperties = {
              disableVisualSelection: row === 1
            };
```

Reply:
Interesting comment addressing look at performance 👍 I wouldn't normally think about it, because it's just a test (as you noticed)... but second option provided by you is really clean, so why not 🙂? I'm changing it within PR and in other connected PRs.

Reply:
Oh, it seems that it may be buggy. I'm checking it.

Reply:
Ok, I've got it. It seems that `disableVisualSelection` set to `false` by `cells` method with new body (before the change it hasn't set value for `disableVisualSelection` in all cases) override `true` value sets by the `columns` property.

Thus, I'm reverting it. I'm suggesting to leave code as it was. For me, the first approach isn't so clear. I don't like a ternary operator.

Reply:
Changed within https://github.com/handsontable/handsontable/pull/7472/commits/fbfec0338d2cbddf621ddcd8562869a2c5ce64bc.